cfme/tests/storage/test_volume.py

```diff
@@ -51,6 +53,27 @@ def volume(appliance, provider):
             msg=str(e)))
 
 
+@pytest.fixture(scope="function")
+def attached_volume(appliance, provider, instance_fixture):
+    volume = create_volume(appliance, provider, from_manager, az=instance_fixture.
+                           vm_default_args["environment"]["availability_zone"], should_assert=True)
+    volume.attach_instance(name=instance_fixture.name, mountpoint='/dev/sdm',
+                           from_manager=from_manager)
+    wait_for(lambda: volume.instance_count == 1, delay=15, timeout=600)
+    yield volume
+
+    try:
+        if volume.exists:
+            if volume.instance_count > 0:
+                volume.detach_instance(name=instance_fixture.name, from_manager=from_manager)
+                wait_for(lambda: volume.instance_count == 0, delay=15, timeout=600)
+            volume.delete(wait=True)
+    except Exception as e:
+        logger.warning("{name}:{msg} Volume deletion - skipping...".format(
```

Comment:
You can use `logger.exception` here; it log the exception and message (with any additional message) for you. Afaik for the logger it is best practice to format strings with `%s` 

https://docs.python.org/3/howto/logging-cookbook.html#formatting-styles

Reply:
I just copied it from volume fixture created by someone else.
changed to logger.exception and deleted string formatting as all the needed info is already in exception.